src/main.h

```diff
@@ -38,7 +38,7 @@ gboolean main_is_realized(void);
 
 typedef struct
 {
-	gboolean	new_instance;
+	gint		new_instance;
```

Comment:
Don't agree with the use of int and values of 1 and -1, make it an enum with sensible names.